### ISSUE TITLE
[bug]: Flux models doesnt work on Mac M2 device. Gets an error like this -> ImportError: The bnb modules are not available. Please install bitsandbytes if available on your platform.

### ISSUE THREAD

**Issue by moheshmohan** [invoke-ai/InvokeAI#6965]
### Is there an existing issue for this problem?

- [X] I have searched the existing issues

### Operating system

macOS

### GPU vendor

Apple Silicon (MPS)

### Version number

5.0.0

### Browser

chrome 129.0.6668.60

### What happened

When i run flux models i get the below error

ImportError: The bnb modules are not available. Please install bitsandbytes if available on your platform.

### What you expected to happen

I expected flux models to run. I have been using flux on same device using other software like diffusionbee

**Comment by colinux** [comment]
I installed `bitsandbytes` manually (`pip3 install bitsandbytes` in the virtual env).
Then there is the error `TypeError: BFloat16 is not supported on MPS`, which lead to me to  https://github.com/bitsandbytes-foundation/bitsandbytes/issues/1020 : port for MPS architecture is still a WIP.
From my understanding we have to wait this to be completed, unless another solution emerges with different quantizations or compression techniques. I tried to import the Flux fp8 checkpoints from DrawThings but they are not compatible.

**Comment by Adreitz** [comment]
As far as I know, you will need to update torch and torchvision to a more recent version to get bfloat16 support on MPS. You can try torch 2.3.1 or a recent nightly build. However, I don't know if this will help you get quantized Flux working, as MPS does not support fp8 at all.

**Comment by Vargol** [comment]
Yep, I'm afraid you need enough memory to run the fp16 version of Flux at the moment as bitsandbtyes doesn't support MacOS. 

And you'll need to upgrade torch in your InvokeAI venv.

You also need a couple of code changes doing to the InvokeAI code (which in theory now depends on what version of torch you upgrade to, with 2.3.1  you need a change a function call that is in theory fixed in the PyTorch nightlies now I not tested it yet).

**Comment by vicento** [comment]
following this. i face to the same problem **ImportError: The bnb modules are not available. Please install bitsandbytes if available on your platform.**

**Comment by cchance27** [comment]
GGUF work on 2.4.1 pytorch, (nightly pytorch break GGUF)... at least in comfy, and it works with all other nodes.

Also their are now MLX nodes as well for comfy to load 4bit versions of flux ... but that has other issues (no compatability with most things yet)

**Comment by smile4yourself** [comment]
I tried v5.5 on my MAC and have similar problems, and while some comments indicate that there may be some FLUX models that may work, I have never succeeded to find one. DrawThings works with FLUX models on the MAC, and I tried using one of their models, moving it to Invoke, but it too did not work.

**Comment by RyPoints** [comment]
It should be possible to route around Bit and Bytes.  I did this for the Kandinsky 3 release to get it working on MPS.  I looked at this a little.  It's mostly in the `flux.py` and `module.py`, but saw some size mismatches on my quick attempts to get it working.  ComfyUI itself is working with FLUX, so one would think Invoke should be able to too.

**Comment by Zanges** [comment]
yo

**Comment by 4-alok** [comment]
Recommend me some other Stable Diffusion models to work with. My M4 Pro struggles with the large 33GB flux schnell model, and I'd prefer something smaller and faster than NF4/F8.

**Comment by pidg** [comment]
t5_bnb_int8_quantized_encoder gets installed as part of the FLUX starter pack. 

If you're a newbie (like me) you might install this starter pack then wonder why it isn't working on your M-series processor.  You then install FLUX Schnell (bfloat16) and still get the bnb error.

The solution seems to be to remove *everything* installed by the starter pack, especially t5_bnb_int8_quantized_encoder, before installing the non-quanitzed FLUX Schnell.

**Comment by smile4yourself** [comment]
your solution worked pidg! This is the first time flux worked on my MAC. It's dredfully slow, as to make it really unusable, but it did work. Drawthings works with flux on the MAC an d creates loras you can use in invoke ai (with other SDXL models...

**Comment by fuzzy76** [comment]
Rephrased: Remove the bnb encoder, remove all models marked as quantized, and keep away from them.

**Comment by alexjyong** [comment]
I'm getting similar issues as well. 

I've had better luck with https://github.com/divamgupta/diffusionbee-stable-diffusion-ui which was designed with macs in mind. 

Not as feature rich though, and unfortunately, there hasn't been any updates since August of last year.  But it gets the job done.

Judging by this message on the Discord, efforts are probably being focused on a newer tool:

<img width="1054" height="197" alt="We are building something amazing. A local tool to easily create and edit high-quality images locally. ( similar to midjourney ).  We are looking for users for early access and feedback. If interested DM me and add a reaction here. In general we are also looking to talk to users, understand their pain points, so that we can build better tools for them. " src="https://github.com/user-attachments/assets/3b580137-7e12-4976-8d88-14090ed000bd" />
